src/EventListener/ChangeChannelListener.php

```diff
@@ -0,0 +1,25 @@
+<?php
+declare(strict_types=1);
+
+namespace AppBundle\EventListener;
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+
+class ChangeChannelListener
+{
+    public function onKernelResponse(ResponseEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+        if (!$event->getRequest()->query->has('change_channel')) {
+            return;
+        }
+
+        $response = $event->getResponse();
+        $newChannel = $event->getRequest()->query->get('change_channel');
```

Comment:
It should not be possible to inject the channel via a request parameter without sanitizing input. 

In the current implementation, if you do `?change_channel=notexists`, it stores the `notexists` string in cookies, and `ProChannelContext ` then returns a non-existing channel, leading to the following error:

```
Return value of AppBundle\Sylius\Channel\ProChannelContext::getChannel() must implement 
interface Sylius\Component\Channel\Model\ChannelInterface, null returned
```

Then it goes into an infinite error 500.

---

At least, the value of `$newChannel` should only allow a finite list of values (`web` or `pro`). We don't want customers to switch to the `app` channel. 
For other values, it should just do nothing.